Add CentOS documentation and improve dockerfiles for UCX (#2537)
NVIDIA/spark-rapids: docs/additional-functionality/rapids-shuffle.md

   
   ---
   
##### CentOS UCX RPM

> The UCX packages for CentOS 7 and 8 are divided into different RPMs. For example, for UCX 1.10.1, 
> fetched from https://github.com/openucx/ucx/releases/download/v1.10.1/ucx-v1.10.1-centos7-mofed5.x-cuda11.0.tar.bz2, 
> is a tarball that contains:

Reviewer: ```suggestion
The UCX packages for CentOS 7 and 8 are divided into different RPMs. For example, UCX 1.10.1 
available at
https://github.com/openucx/ucx/releases/download/v1.10.1/ucx-v1.10.1-centos7-mofed5.x-cuda11.0.tar.bz2
contains:
```

```
ucx-devel-1.10.1-1.el7.x86_64.rpm   
ucx-debuginfo-1.10.1-1.el7.x86_64.rpm
ucx-1.10.1-1.el7.x86_64.rpm         
ucx-cuda-1.10.1-1.el7.x86_64.rpm
ucx-rdmacm-1.10.1-1.el7.x86_64.rpm  
ucx-cma-1.10.1-1.el7.x86_64.rpm
ucx-ib-1.10.1-1.el7.x86_64.rpm
```

For a setup without RoCE or Infiniband networking, the only packages required are: 

```
ucx-1.10.1-1.el7.x86_64.rpm, and 
ucx-cuda-1.10.1-1.el7.x86_64.rpm. 
```

> If the accelerated networking is available, the package list is: 

Reviewer: ```suggestion
If accelerated networking is available, the package list is: 
```

```
ucx-1.10.1-1.el7.x86_64.rpm, 
ucx-cuda-1.10.1-1.el7.x86_64.rpm,
ucx-rdmacm-1.10.1-1.el7.x86_64.rpm, and
ucx-ib-1.10.1-1.el7.x86_64.rpm.
```

---
**NOTE:**

> The CentOS RPM requires CUDA installed via RPMs to satisfy its dependencies.

Reviewer: Is there a useful link we can place here to direct users on how to install CUDA via RPM?  Or maybe point to the CUDA toolkit front-page and walk them through how to navigate that site to instructions on installing via RPM?

Author: Sure thing, will add.

Author: added a specific example here: https://github.com/NVIDIA/spark-rapids/pull/2537/commits/d132c7f1730a0fb9a293dbd436b3afd7a304e17a, though its debatable whether it "fits" the UCX doc.

Within the Docker container we need to install UCX and its requirements. These are Dockerfile
examples for Ubuntu 18.04:

The following are examples of a Docker containers with UCX 1.10.1 and cuda-11.0 support. We

> break down the examples by OS, and further on whether RDMA-capable hardware is available:

Reviewer: ```suggestion
The following are examples of Docker containers with UCX 1.10.1 and cuda-11.0 support. The
examples are divided by OS type and RDMA-capable hardware availability.
```

Author: done

> ##### Ubuntu

Reviewer: Does it make sense to present this as a table rather than subsections?

Within the Docker container we need to install UCX and its requirements. These are Dockerfile
examples for Ubuntu 18.04:

> The following are examples of Docker containers with UCX 1.10.1 and cuda-11.0 support. The 
> examples are divided by OS type and RDMA-capable hardware availability.

Reviewer: Now that it's in table form, the table headers makes the second sentence redundant.
```suggestion
The following are examples of Docker containers with UCX 1.10.1 and cuda-11.0 support.
```

Author: Thanks good point. I've added this one, which unfortunately dismissed your approval.
